src/listeners/github/handlers/workflow-run-handler-v2.js

```diff
@@ -81,26 +81,65 @@ const handleCdpCreateWorkflows = async (db, message) => {
 
   try {
     const repoName = message.workflow_run?.name // we repurpose the name to track name of repo its creating
-    const updateStatus = updateRepositoryStatus(db, repoName)
     const status = findByRepoName(db, repoName)
 
-    if (status !== null) {
-      const workflowStatus = normalizeStatus(
-        message.action,
-        message.workflow_run?.conclusion
-      )
-      logger.info(
-        `updating createRepository status fro ${repoName} to ${workflowStatus}`
-      )
+    if (status === null) {
+      return
+    }
 
-      await updateStatus({
-        status: workflowStatus,
-        'createRepository.status': workflowStatus
-      })
+    const workflowStatus = normalizeStatus(
+      message.action,
+      message.workflow_run?.conclusion
+    )
+
+    logger.info(
+      `attempting to update createRepository status for ${repoName} to ${workflowStatus}`
+    )
+
+    // make sure statuses can only be progressed forward, not back (request -> in-progress -> success/failure)
+    // this can happen if the github events arrives in the wrong order
+    let dontOverwrite = []
+    switch (workflowStatus) {
+      case statuses.requested:
+        dontOverwrite = [
+          statuses.success,
+          statuses.failure,
+          statuses.inProgress
+        ]
+        break
+      case statuses.inProgress:
+        dontOverwrite = [statuses.success, statuses.failure]
+        break
+      case statuses.success:
+      case statuses.failure:
+        dontOverwrite = []
+    }
 
-      // TODO combine this and new update-repository-status work
-      await updateOverallStatus(db, repoName)
+    const updateResult = await db.collection('status').updateOne(
+      {
+        repositoryName: repoName,
+        'createRepository.status': { $nin: dontOverwrite }
+      },
+      {
+        $set: {
+          status: workflowStatus,
+          'createRepository.status': workflowStatus
+        }
+      }
+    )
+
+    if (updateResult.matchedCount > 0) {
+      logger.info(
+        `set ${repoName} createRepository status to ${workflowStatus}`
+      )
+    } else {
+      logger.warn(
+        `did NOT set ${repoName} createRepository to ${workflowStatus}`
+      )
```

Comment:
maybe also expose why in the logs?:
```
did NOT set ${repoName} createRepository to ${workflowStatus} because createRepository.status is ${createRepositoryStatus}
```